providers/google/util.go

```
@@ -85,17 +97,139 @@ func convertGoogleResponse(resp *genai.GenerateContentResponse, model string) (*
+		"thoughts": thoughts, "tool_use": toolUse, "total": total,
+	} {
+		if count < 0 {
+			return llm.Usage{}, fmt.Errorf("invalid negative Gemini %s token count: %d", name, count)
+		}
+	}
+	if cached > prompt {
+		return llm.Usage{}, fmt.Errorf("invalid Gemini cached token count %d above prompt count %d", cached, prompt)
+	}
+	expectedTotal := prompt + candidates + thoughts + toolUse
+	if total != expectedTotal {
+		return llm.Usage{}, fmt.Errorf(
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)